### DISCUSSION TITLE
Is there a way we can dynamically update the atoms just by passing the name of the atom and next value to update it?

### DISCUSSION THREAD

**Discussion by bhushan-patil-official** [pmndrs/jotai#1551] [Q&A]
So we have multiple atoms, what we were trying to do is update any atom by passing its name in the function and updating that atom's value. Like this.

```
// Defined atom
const firstAtom = atom<number>(0);

// function to dynamically update the atom.
const handleStateMgr = (msg: { atomName: any, value: any }) => {
		atom(
			(get) => get(msg.atomName),
			(get, set) => {
				const nextValue =	typeof msg.value === "function"
					? (msg.value as (prev: any) => any)(get(msg.atomName))
					: msg.value;
				set(msg.atomName, nextValue);
			},
		);
	};

handleStateMgr({firstAtom, (current: number) => current + 1})
```

Was trying to do something like this but not able to achieve it. We are open to use a hook inside it but not sure how to define the hook and accept the atom name in it.

Thankyou in advance

**Comment by edongashi** [comment]
We have been discussing something similar in #1543. All solutions are somewhat hacky and against the idea of the library.

Basically you use a write atom to run custom code:

```js
const atomSetter = atom(null, (get, set, [atom, update]) => set(atom, update));

function SomeComponent() {
  const handleStateMgr = useSetAtom(atomSetter);

  ...
  handleStateMgr([firstAtom, current => current + 1]);
  ...
}
```

Also check out [useAtomCallback](https://jotai.org/docs/utils/use-atom-callback).

**Comment by dai-shi** [comment]
Apart from #1543, one canonical solution at this moment is `useAtomCallback`.
Another is we have an experimental `unstable_createStore`, which will be exposed in the new API #1514.
